scripts/suite_objects.py

```diff
@@ -444,12 +444,17 @@ def add_call_list_variable(self, newvar, exists_ok=False,
                 newvar = oldvar.clone(subst_dict, source_name=self.name,
                                       source_type=stype, context=self.context)
             # end if
+            stdname = newvar.get_prop_value('standard_name')
+            vtype = newvar.get_prop_value('type')
+            dimensions = newvar.get_prop_value('dimensions')
             self.call_list.add_variable(newvar, self.run_env,
                                         exists_ok=exists_ok,
                                         gen_unique=gen_unique,
                                         adjust_intent=True)
             # We need to make sure that this variable's dimensions are available
             for vardim in newvar.get_dim_stdnames(include_constants=False):
+                if vardim == '':
```

Comment:
vardim can be empty if the variable is allocatable. I added a check only allow this when the variable is allocatable and to otherwise throw an error

```
for vardim in newvar.get_dim_stdnames(include_constants=False):
    # Unnamed dimensions are ok for allocatable variables
    if vardim == '' and newvar.get_prop_value('allocatable'):
        continue
    elif vardim == '':
        emsg = f"{self.name}: Cannot have unnamed/empty string dimension"
        raise ParseInternalError(emsg.format(self.name,
                                             vardim, stdname))
    # end if
...
```